build/teamcity/internal/cockroach/build/ci/open-new-bazel-builder-image-pr.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -xeuo pipefail
+
+# This script expects the following env vars to be set:
+#   BRANCH [branch name to be created in fork]
+#   VERSION [new version to echo into build/.bazelbuilderversion]
+#   GH_TOKEN [github token]
+
+dir="$(dirname $(dirname $(dirname $(dirname $(dirname $(dirname "${0}"))))))"
+source "$dir/teamcity-common-support.sh"
+
+git_ssh() {
+  # $@ passes all arguments to this function to the command
+  GIT_SSH_COMMAND="ssh -i /tmp/.cockroach-teamcity-key" git "$@"
+}
+
+# Configure git.
+git config --global user.email "teamcity@cockroachlabs.com"
+git config --global user.name "cockroach-teamcity"
+configure_git_ssh_key
+mv .cockroach-teamcity-key /tmp/.cockroach-teamcity-key
+
+git_ssh clone "ssh://git@github.com/cockroachdb/cockroach.git" "/tmp/cockroach" && cd "/tmp/cockroach"
```

Comment:
Can you use a temp directory (`mktemp -d`) instead of hardcoded `/tmp/cocroach` for a few reasons:
* harder to predict (and attack)
* no need to rely on absence of the directory - you would need to make sure it doesn't exist or empty before cloning - otherwise git fails.

Also, can you `rm -rf` that directory, so we clean after it's done. Maybe add all needed cleanup command to a function and `trap` it? But we should be careful using `trap` - IIRC there is one already sourced, and only one can be used.

Reply:
Moved the clone and key to a temp dir which gets cleaned at the end